tests/unit/lms/views/admin/application_instance_test.py

```diff
@@ -381,6 +381,30 @@ def test_update_instance_saves_ai_settings(
         )
         assert application_instance.settings.get(setting, sub_setting) == expected
 
+    @pytest.mark.parametrize(
+        "key,secret,update",
+        [
+            ("key", "secret", True),
+            ("key", "", False),
+            ("", "secret", False),
+            ("", "", False),
+        ],
+    )
+    def test_update_instance_canvas_api_details(
+        self, pyramid_request, application_instance_service, views, key, secret, update
+    ):
+        pyramid_request.params = {"developer_key": key, "developer_secret": secret}
+        views.update_instance()
+
+        if update:
+            application_instance_service.set_canvas_api_details.assert_called_once_with(
+                application_instance_service.get_by_id.return_value,
+                key,
+                secret,
+            )
+        else:
+            application_instance_service.set_canvas_api_details.assert_not_called()
```

Comment:
```suggestion
    @pytest.mark.parametrize("key,", (" ", " key "))
    @pytest.mark.parametrize("secret", (" ", " secret "))
    def test_update_instance_canvas_api_details(
        self, pyramid_request, application_instance_service, views, key, secret
    ):
        pyramid_request.params = {"developer_key": key, "developer_secret": secret}

        views.update_instance()

        application_instance_service.set_canvas_api_details.assert_called_once_with(
            application_instance_service.get_by_id.return_value,
            developer_key=key.strip() if key else '',
            developer_secret=secret.strip() if secret else '',
        )
```

This isn't great, but it's the best I could come up with.